Update typespec-ts generator references to Azure/typespec-azure (#38807)
Azure/azure-sdk-for-js: sdk/core/core-xml/README.md

Reviewer: This package is not used by the typespec-ts emitter either.

# Azure Core XML client library for JavaScript

> This library is primarily intended to be used in code generated by the [TypeSpec](https://typespec.io) [`@azure-tools/typespec-ts`](https://github.com/Azure/typespec-azure/tree/main/packages/typespec-ts) emitter for APIs that require parsing XML payloads.

Reviewer: ```suggestion
This library is primarily intended to be used in code generated by [AutoRest](https://github.com/Azure/Autorest) and [`autorest.typescript`](https://github.com/Azure/autorest.typescript) for APIs that require parsing XML payloads.
```
